sql/sql_parse.cc

```diff
@@ -2174,6 +2174,7 @@ bool dispatch_command(THD *thd, const COM_DATA *com_data,
       thd->cleanup_connection();
 
 #ifdef WITH_WSREP
+      thd->wsrep_client_thread = true;
```

Comment:
cleanup_connection() is called in two cases: COM_CHANGE_USER and COM_RESET_CONNECTION  . Do we have the same problem in the 2nd case as well?
Maybe it would be better to set wsrep_client_thread = true after THD::cleanup() in cleanup_connection() ?

Reply:
Yes. You are right. COM_RESET_CONNECTION  (\x) is also affected.

``` 
Thread 40 "connection" hit Breakpoint 1, dispatch_command (thd=0x7f0fec002090, com_data=0x7f1004785b40, 
    command=COM_RESET_CONNECTION) at /home/venki/work/pxc/80/sql/sql_parse.cc:2113
2113          wsrep_after_command_ignore_result(thd);
(gdb) n
2114          wsrep_close(thd);
(gdb) n
2116          thd->cleanup_connection();
(gdb) p thd->wsrep_client_thread
$4 = true
(gdb) n
2118          wsrep_open(thd);
(gdb) p thd->wsrep_client_thread
$5 = false
 ```
 

> Maybe it would be better to set wsrep_client_thread = true after THD::cleanup() in cleanup_connection() ? 

IMO removing the `wsrep_client_thread = false` line  from  `THD::cleanup()` and placing inside `end_connection()` would be a better solution. We set `wsrep_client_thread = true` inside `thd_prepare_connection()`. So, moving the cleanup code to inside `end_connection()` would be reasonable in this case.